bentoml/_internal/utils/http.py

```diff
@@ -0,0 +1,36 @@
+import typing as t
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import starlette.responses
+
+
+def finalize_http_response(response: "starlette.responses.Response"):
+    body = getattr(response, "body", None)
+    if body is not None and not (
+        response.status_code < 200
+        or response.status_code in (204, 304)  # type:ignore (starlette)
+    ):
+        found_indexes: t.List[int] = []
+        update = True
+        for idx, (item_key, item_value) in enumerate(response.raw_headers):
+            if item_key == b"content-length":
+                if item_value == b"0":
+                    found_indexes.append(idx)
+                else:
+                    update = False
+
+        if update:
+            for idx in reversed(found_indexes[1:]):
+                del response.raw_headers[idx]
+
+            new_context_len = str(len(body)).encode("latin-1")
```

Comment:
what happens if the body is not using latin-1 encoding?

Reply:
`body` is bytes here; this is encoding the content-length.